chatbot/commandcenter/commander/commander.py

```diff
@@ -7,98 +7,79 @@
 # important! this is where we discover all of the commands in the command directory.
 # if this isn't called, Command.__subclasses__() is empty.
 from ..commands import *
+from .builtin import *
 
 from ..command import Command
 from ..eventpackage import EventPackage
 
-# dummy classes of the built-in commands so that they are compatible with commands like $help or $info
-
-class HelpCommand(Command):
-    def __init__(self):
-        self.name = "$help" # this is required in order for the command to run!
-        self.help = "$help - Learn about commands. Usage: `$help` or `$help <command (including the dollar sign)>`"
-        self.author = "spacedog"
-        self.last_updated = "Sept. 28, 2018"
-
-    def run(self, event_pack: EventPackage):
-        return "This command is built in to the commander. You should never see this text. Uh oh."
-
-class InfoCommand(Command):
-    def __init__(self):
-        self.name = "$info" # this is required in order for the command to run!
-        self.help = "$info - Display info about a command. Usage: `$info <command (including the dollar sign)>`"
-        self.author = "spacedog"
-        self.last_updated = "Sept. 28, 2018"
-
-    def run(self, event_pack: EventPackage):
-        return "This command is built in to the commander. You should never see this text. Uh oh."
-
 class Commander:
-    def __init__(self):
+    def __init__(self, prefix = "$"):
         self.commands = {}
+        self.prefix = prefix
+
+        print("Prefix: " + self.prefix)
 
         for command_class in Command.__subclasses__():
             command = command_class()
             self.commands[command.get_name()] = command
-            print("Loaded command: " + command.get_name())
+
+        print("Commander loaded with commands:")
+        print(self.get_help_all())
 
     def run_command(self, command_string, event_pack: EventPackage):
-        print("Commander: running command: {0}".format(command_string))
+        # convert command string to use $ as prefix in order to work with the internal syntax
+        command_string_normalized = command_string.replace(self.prefix, "$")
+
+        print("Commander: running command: {0}".format(command_string_normalized))
```

Comment:
This has to do with changing the command prefix. the internals still use $ as the prefix to make defining things easier. This bit of code changes whatever custom prefix is being used to $ so it can be easily used internally.